### DISCUSSION TITLE
Compound component dot notation

### DISCUSSION THREAD

**Discussion by davidduncandev** [radix-ui/themes#178] [Q&A] [answered]
I've been wondering about dot notation versus named exports? Is there a specific reason Radix adopts dot notation as opposed to named exports for all component parts? Can it impact tree-shaking and any know compatibility issues with NextJS?

**Comment by vladmoroz** [comment, marked as answer]
Radix Themes exports both the dot notation version and individual parts for every component. Our personal preference when building UIs is to use the dot notation, but Next.js server components don't support that well yet, so we'll have to see how that pans out over time. As for tree-shaking, for multi-part components you usually need all or almost all of the parts anyway. We do have some tree-shaking issues, but that's not related to dot notation and will be fixed soon. 

Radix Primitives provides each primitive as a separate package, and how your import them defines whether you get dot notation or the individual parts of a primitive.
